### PR TITLE
Rename tel into phone to ensure compatibility with Symfony 3.4 form_div_layout

### PR DESCRIPTION
**Pull request by JusteLeblanc**
I updated the library to make it compatible with last form_div_layout of symfony 3.4

To fix the issue https://github.com/misd-service-development/phone-number-bundle/issues/83

### CODE REVIEW

File: Form/Type/PhoneNumberType.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
-        $view->vars['type'] = 'tel';
+        $view->vars['type'] = 'phone';
```
Comment (toooni) [inline Form/Type/PhoneNumberType.php]: @JusteLeblanc The html5 specification is `tel` not `phone` (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel)

**Comment by nealio82** [comment]
I suspect the wrong line was changed?

leaving ln 120 as `$view->vars['type'] = 'tel';` but changing ln 191 to `return 'phone';` fixes the exception for me

File: Form/Type/PhoneNumberType.php (hunk 1 above)
Comment (JusteLeblanc) [reply Form/Type/PhoneNumberType.php]: Thanks ! I'll push changes today

**Comment by teohhanhui** [comment]
Why not just rename it to `misd_phone_number`?

**Comment by giovkanata** [comment]
@rh389  Can we merge it?

**Comment by JusteLeblanc** [comment]
Hello @rh389 
I need help on this PR: 

1. Tests are red as they are not passing anymore on PHP HHVM, should we stop to ensure compatibility with PHP HHVM ?
2. I renamed `tel_widget` into `phone_widget` to ensure compatibility with Symfony 3.4+ `form_div_layout`. Other ideas suggested by reviewers: `custom_tel_widget` or `misd_phone_number`. My favorite one is `custom_tel_widget`. Any preference?

Thanks!

**Comment by teohhanhui** [comment]
I really think it should use proper namespacing, so `misd_phone_number`

**Comment by rh389** [comment]
This is a tricky one - in principle I like namespacing but I think there's a decent argument for `phone_number_widget`, for consistency with `phone_number_helper`, `phone_number_format` and `phone_number_of_type`. 

Is there any reason to namespace one when the others are not?

**Comment by teohhanhui** [comment]
If it's a Twig filter, I could understand why it's not namespaced, as it'd be too verbose to write in templates. But otherwise, we should namespace everywhere.

**Comment by JusteLeblanc** [comment]
> This is a tricky one - in principle I like namespacing but I think there's a decent argument for `phone_number_widget`, for consistency with `phone_number_helper`, `phone_number_format` and `phone_number_of_type`.
> 
> Is there any reason to namespace one when the others are not?

I agree with `phone_number_widget` for consistency reason. cc @teohhanhui  I updated the PR. 

@rh389 could we merge it as tests are red because of PHP5.3 and PHP HHVM builds? Other builds are 👍

**Comment by JusteLeblanc** [comment]
@rh389 Any feed-back on this PR ? Would be great to merge it :)

**Comment by rh389** [comment]
Looks good, just haven’t had time try it out and to update the docs - if you’re happy to update the readme that’d be great, otherwise I’ll merge when I get chance. Cheers for the work👍🏻

**Comment by JusteLeblanc** [comment]
> Looks good, just haven’t had time try it out and to update the docs - if you’re happy to update the readme that’d be great, otherwise I’ll merge when I get chance. Cheers for the work👍🏻

I've updated Changelog. I've found nothing to update in the Readme.
I try out the bundle, it works fine. 😸 
I removed PHP 5.3 on which build failed as it's not maintained since August 2014. 🙈 

All tests are green except PHP hhvm ones. I will have a look on it.

@rh389 Sounds good for you? 😇

**Comment by JusteLeblanc** [comment]
I specified Symfony version for PHP hhvm build as [Symfony 4 ended HHVM support](https://symfony.com/blog/symfony-4-end-of-hhvm-support). All tests are green! ✅

@rh389 It's ready to be merged 😸

**Comment by JusteLeblanc** [comment]
> I specified Symfony version for PHP hhvm build as [Symfony 4 ended HHVM support](https://symfony.com/blog/symfony-4-end-of-hhvm-support). All tests are green! ✅
> 
> @rh389 It's ready to be merged 😸

@rh389 is there anything I could do to help you merge this PR?

**Comment by rh389** [comment]
Many thanks for your work and patience @JusteLeblanc - merged!

**Pull request merged**